omod/src/main/java/org/openmrs/module/aijar/web/controller/AppointmentCountController.java

```diff
@@ -0,0 +1,63 @@
+
+package org.openmrs.module.aijar.web.controller;
+
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.AdministrationService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+public class AppointmentCountController extends BaseRestController {
+
+    AdministrationService administrationService = Context.getAdministrationService();
+
+    //@RequestMapping(method = RequestMethod.GET)
+    @RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/ugandaemr/appointmentcount",method = RequestMethod.GET)
+    @ResponseBody
+    public SimpleObject getNumberOfPatients(
+            @RequestParam(required = true, value = "nextAppointmentDate") Date nextAppointmentDate,
+            @RequestParam(value = "encounterType", defaultValue = "") String encounterType) {
+
+        SimpleObject response = new SimpleObject();
+
+        SimpleDateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+        String returnDate = formatter.format(nextAppointmentDate);
+        List patients = new ArrayList();
+
+        if (StringUtils.isEmpty(encounterType)) {
+            String sql = "SELECT obs.person_id FROM obs WHERE obs.concept_id=5096 AND obs.value_datetime='" + returnDate
+                    + "' AND obs.voided = 0";
+            patients = administrationService.executeSQL(sql, true);
+
+        } else {
+            String[] encounterT = encounterType.split(",");
+            for (String e : encounterT) {
+                String sql = "SELECT obs.person_id FROM obs WHERE obs.concept_id=5096 AND obs.value_datetime='"
+                        + returnDate
+                        + "' AND obs.voided = 0 AND obs.encounter_id IN ( SELECT encounter.encounter_id FROM encounter INNER JOIN encounter_type ON encounter.encounter_type = encounter_type.encounter_type_id    WHERE encounter_type.uuid IN ('"
+                        + e + "'))";
+                List patientsinEncounter = administrationService.executeSQL(sql, true);
+                patients.addAll(patientsinEncounter);
+            }
+
+        }
+        response.add("appointmentdate", returnDate);
+        response.add("encountertype", encounterType);
+        response.add("appointmentcount", patients.size());
```

Comment:
@reagan-meant This may be very slow, so in the query run a count and return a single column that is used here

Reply:
Not sure I get what you mean...What would be the returnType ...and @ line 58 I have the size.

Reply:
@reagan-meant The patients.size() will load all the patient data from the database so that you can count how many are there, your query needs to be SELECT COUNT(obs.person_id) as appointmentcount so that you return that value already computed from the database. Trust me on this one

Reply:
@ssmusoke I didnt doubt..Just hadnt got what you meant.. Your suggestion makes alot of sense...I got confused coz it seemed as though you meant the code from 56 to 58 was to be in one column.